lib/node_modules/@stdlib/blas/ext/last-index-of/lib/main.js

```diff
@@ -0,0 +1,203 @@
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var hasOwnProp = require( '@stdlib/assert/has-own-property' );
+var isPlainObject = require( '@stdlib/assert/is-plain-object' );
+var isInteger = require( '@stdlib/assert/is-integer' ).isPrimitive;
+var isndarrayLike = require( '@stdlib/assert/is-ndarray-like' );
+var broadcastScalar = require( '@stdlib/ndarray/base/broadcast-scalar' );
+var maybeBroadcastArray = require( '@stdlib/ndarray/base/maybe-broadcast-array' );
+var getDType = require( '@stdlib/ndarray/dtype' );
+var getShape = require( '@stdlib/ndarray/shape' );
+var getOrder = require( '@stdlib/ndarray/order' );
+var format = require( '@stdlib/string/format' );
+var defaults = require( '@stdlib/ndarray/defaults' );
+var nonCoreShape = require( './non_core_shape.js' );
+var base = require( './base.js' );
+
+
+// VARIABLES //
+
+var DEFAULT_DTYPE = defaults.get( 'dtypes.integer_index' );
+
+
+// MAIN //
+
+/**
+* Returns the last index of a specified search element along an ndarray dimension.
+*
+* @param {ndarrayLike} x - input ndarray
+* @param {(ndarrayLike|*)} searchElement - search element
+* @param {(ndarrayLike|integer)} [fromIndex] - index from which to begin searching
+* @param {Options} [options] - function options
+* @param {IntegerArray} [options.dim=-1] - dimension over which to perform operation
+* @param {boolean} [options.keepdims=false] - boolean indicating whether the reduced dimensions should be included in the returned ndarray as singleton dimensions
+* @param {string} [options.dtype] - output ndarray data type
+* @throws {TypeError} first argument must be an ndarray-like object
+* @throws {TypeError} second argument must be either an ndarray-like object or a scalar value
+* @throws {TypeError} third argument must be either an ndarray-like object or an integer
+* @throws {TypeError} options argument must be an object
+* @throws {RangeError} dimension index must not exceed input ndarray bounds
+* @throws {RangeError} first argument must have at least one dimension
+* @throws {Error} must provide valid options
+* @returns {ndarray} output ndarray
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+* var ndarray2array = require( '@stdlib/ndarray/to-array' );
+* var ndarray = require( '@stdlib/ndarray/ctor' );
+*
+* // Create a data buffer:
+* var xbuf = new Float64Array( [ 1.0, 2.0, -3.0, 2.0, -5.0, 6.0 ] );
+*
+* // Define the shape of the input array:
+* var sh = [ 2, 3 ];
+*
+* // Define the array strides:
+* var sx = [ 3, 1 ];
+*
+* // Define the index offset:
+* var ox = 0;
+*
+* // Create an input ndarray:
+* var x = new ndarray( 'float64', xbuf, sh, sx, ox, 'row-major' );
+*
+* // Perform operation:
+* var out = lastIndexOf( x, 2.0 );
+* // returns <ndarray>
+*
+* var arr = ndarray2array( out );
+* // returns [ -1, 0 ]
```

Comment:
```suggestion
* // returns [ -1, 0 ]
```

This isn't correct either.

```
[
  1.0, 2.0, -3.0,
  2.0, -5.0, 6.0
]
```

Over the last dimension means across the columns. In which case, I'd expect `[ 1, 0 ]`.